hail/src/main/scala/is/hail/expr/types/physical/PCanonicalLocus.scala

```diff
@@ -0,0 +1,104 @@
+package is.hail.expr.types.physical
+import is.hail.variant.ReferenceGenome
+
+import is.hail.annotations._
+import is.hail.asm4s.{Code, coerce}
+import is.hail.backend.BroadcastValue
+import is.hail.expr.ir.EmitMethodBuilder
+import is.hail.expr.types.virtual.TLocus
+import is.hail.utils._
+import is.hail.variant._
+
+
+object PCanonicalLocus {
+  def apply(rg: ReferenceGenome): PLocus = PCanonicalLocus(rg.broadcastRG)
+
+  def apply(rg: ReferenceGenome, required: Boolean): PLocus = PCanonicalLocus(rg.broadcastRG, required)
+
+  def representation(required: Boolean = false): PStruct = PStruct(
+    required,
+    "contig" -> PString(required = true),
+    "position" -> PInt32(required = true))
+
+  def schemaFromRG(rg: Option[ReferenceGenome], required: Boolean = false): PType = rg match {
+    case Some(ref) => PCanonicalLocus(ref)
+    case None => representation(required)
+  }
+}
+
+final case class PCanonicalLocus(rgBc: BroadcastRG, required: Boolean = false) extends PLocus {
+    def rg: ReferenceGenome = rgBc.value
```

Comment:
where does PCanonicalLocus define its representation?

Reply:
https://github.com/hail-is/hail/pull/7712/files#diff-1fd93ff173c6dbb75346aecc5ab00242R98

Reply:
ah! missed that.

Reply:
Left there because schemaFromRG is used (and as a static method by the caller)

Reply:
Caller is BGenRDD: https://github.com/hail-is/hail/pull/7712/files#diff-a31345744b19979a03a5e0c8ef90a5a1R103